api/envoy/api/v2/route/route.proto

```diff
@@ -921,6 +921,10 @@ message HeaderMatcher {
     //   "-1somestring"
     envoy.type.Int64Range range_match = 6;
   }
+
+  // If specified, header match will be performed based on if the header value is *not* the value
+  // specified or if the header is not present. Defaults to false.
```

Comment:
The fact that the inversion match will match if the header is not present at all might be confusing to people. Thoughts? I wonder if that should be the case or if we should have an explicit match type for "present" or something like that?

Reply:
It bothers me, especially when looking at tests for large numbers of routes and seeing how many requests the inverse one like this matches.  The original request was for this behavior, and I didn't want to overthink it on my second PR. :-)

I'll add a "present" match type as well to make this more clear as you suggest.